editor/src/components/navigator/navigator.tsx

```diff
@@ -323,3 +328,53 @@ export const NavigatorComponent = React.memo(() => {
   )
 })
 NavigatorComponent.displayName = 'NavigatorComponent'
+
+function useDefaultCollapsed() {
+  const dispatch = useDispatch()
+
+  // keep track of the previously collapsed element paths, or they will be re-collapsed
+  type PreviouslyCollapsed = { [path: string]: boolean }
+  const [previouslyCollapsed, setPreviouslyCollapsed] = React.useState<PreviouslyCollapsed>({})
+
+  const elementNamesByElementPath = useEditorState(
+    Substores.metadata,
+    (store) => {
+      const elements = Object.values(store.editor.jsxMetadata)
+      return elements.reduce((acc, element) => {
+        if (isRight(element.element) && isJSXElement(element.element.value)) {
+          acc[EP.toString(element.elementPath)] = element.element.value.name.baseVariable
+        }
+        return acc
+      }, {} as { [key: string]: string })
+    },
+    'useDefaultCollapsed elementNamesByElementPath',
+  )
+
+  const collapsedPaths = useEditorState(
+    Substores.navigator,
+    (store): Set<string> => new Set(store.editor.navigator.collapsedViews.map(EP.toString)),
+    'useDefaultCollapsed collapsedPaths',
+  )
+
+  React.useEffect(() => {
+    const newCollapsedViews = Object.keys(elementNamesByElementPath).filter((path) => {
+      return (
+        // the collapsed views don't already contain the element
+        !collapsedPaths.has(path) &&
+        // the collapsed view wasn't already collapsed before
+        previouslyCollapsed[path] == null &&
+        // the element is a `head`
+        elementNamesByElementPath[path] === 'head'
```

Comment:
won't that result only in the "head" path (i.e an array with max length of 1)?

Reply:
If you mean it will only contain the `<head>` path of `<head><foo /><bar /></head>` (and not `foo` and `bar`), yes but that's desired.
Or am I misunderstanding and you mean something else?